### PR TITLE
Pass whole data-set

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
BC break: Pass whole data-set to Interpreter, Runner, Filter and Setter. This is a BC break cause it adds a new parameter to FilterInterface, InterpreterInterface, RunnerInterface and SetterInterface. But the use for this is more important than the BC break. This is very important especially in Runner Configurations to process Objects based on the whole data set, for example when generating Variants.

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
@kubaplas Since you are a very active user of this bundle, could you please do me a favor and review this PR? It is a massive BC break, but I think the result overweighs the BC break.

File: src/ImportDefinitionsBundle/Importer/Importer.php [hunk 1]
```diff
@@ -215,56 +215,55 @@ private function getData(DefinitionInterface $definition, $params)
...
-                $this->eventDispatcher->dispatch('import_definition.progress', new ImportDefinitionEvent($definition));
             }
+
+            $this->eventDispatcher->dispatch('import_definition.progress', new ImportDefinitionEvent($definition));
         }
     }
 
     /**
      * @param DefinitionInterface $definition
      * @param $data
```
Comment (kkarkus) [inline src/ImportDefinitionsBundle/Importer/Importer.php]: missing types

File: src/ImportDefinitionsBundle/Importer/Importer.php [hunk 2]
```diff
@@ -215,56 +215,55 @@ private function getData(DefinitionInterface $definition, $params)
...
+     * @param $dataSet
      * @param $params
      * @param null $filter
```
Comment (kkarkus) [inline src/ImportDefinitionsBundle/Importer/Importer.php]: probably type mismatch

File: src/ImportDefinitionsBundle/Importer/Importer.php [hunk 3]
```diff
@@ -332,11 +331,12 @@ private function importRow(DefinitionInterface $definition, $data, $params, $fil
      * @param Mapping $map
      * @param $value
```
Comment (kkarkus) [inline src/ImportDefinitionsBundle/Importer/Importer.php]: missing types

File: src/ImportDefinitionsBundle/Importer/Importer.php [hunk 4]
```diff
@@ -332,11 +331,12 @@ private function importRow(DefinitionInterface $definition, $data, $params, $fil
...
      * @param $data
+     * @param $dataSet
      * @param DefinitionInterface $definition
      * @param $params
      * @param RunnerInterface $runner
```
Comment (kkarkus) [inline src/ImportDefinitionsBundle/Importer/Importer.php]: type mismatch

File: src/ImportDefinitionsBundle/Runner/SetterRunnerInterface.php [hunk 5]
```diff
@@ -25,9 +25,10 @@
      * @param Mapping $map
      * @param $value
```
Comment (kkarkus) [inline src/ImportDefinitionsBundle/Runner/SetterRunnerInterface.php]: missing types

**Comment by dpfaffenbauer** [comment]
thats a good idea, will do that.

**Comment by dpfaffenbauer** [comment]
@kubaplas just did, can you check again?

**Comment by kubaplas** [comment]
Great! My interpreters and setters should still work :D

**Pull request merged**